### PR TITLE
fix(gatsby-dev-cli): Correctly catch 404s from NPM when a package hasn't been published yet

### PR DESCRIPTION
**Pull request by KyleAMathews**
Can't use gatsby-dev-cli with new package otherwise

### CODE REVIEW

File: packages/gatsby-dev-cli/src/utils/check-deps-changes.js [hunk 1]
```diff
@@ -61,15 +61,13 @@ exports.checkDepsChanges = async ({
...
             if (response && response.statusCode === 200) {
-              return resolve(JSON.parse(response.body))
+              resolve(JSON.parse(response.body))
             }
-
-            return reject(error)
-          }
-        )
+          })
+          .catch(e => reject(e))
```
Comment (pieh) [inline packages/gatsby-dev-cli/src/utils/check-deps-changes.js:70]: `.then` handler can only have single argument - so this should be:

```
got(`https://unpkg.com/${packageName}/package.json`)
  .then(response => {
    if (response && response.statusCode === 200) {
      return resolve(JSON.parse(response.body))
    }

    return reject(new Error(`No response or non 200 code`))
  })
  .catch(e => reject(e))
```

Seems like migration in https://github.com/gatsbyjs/gatsby/pull/26653 was never actually tested?

Comment (pvdz) [reply packages/gatsby-dev-cli/src/utils/check-deps-changes.js:70]: Considering this is inside an `async` function it should be something like

```js
const response = await got(`https://unpkg.com/${packageName}/package.json`)
if (response?.statusCode !== 200) {
  throw new Error(`No response or non 200 code`)
}
resolve(JSON.parse(response.body))
```

Comment (pvdz) [reply packages/gatsby-dev-cli/src/utils/check-deps-changes.js:70]: (unless `got` does not return a proper promise-like?)

Comment (pieh) [reply packages/gatsby-dev-cli/src/utils/check-deps-changes.js:70]: Ah, yeah, you are probably right and `await` will simplify things - I mostly zoomed in on double args for `.then` :)

Comment (KyleAMathews) [reply packages/gatsby-dev-cli/src/utils/check-deps-changes.js:70]: >Seems like migration in #26653 was never actually tested?

Yeah — this isn't a common scenario.

Will change to await 👍

**Review by pieh — APPROVED** [review]
Thanks

**Pull request merged**